app/src/main/java/com/num/controller/managers/DnsManager.java

```diff
@@ -0,0 +1,152 @@
+        dnses = new ArrayList<>();
+
+    }
+
+    /**
+     * Retrieve Dns Targets and fire Dns threads for resolving names
+     *
+     */
+    public void execute() {
+        //API mobilyzer = API.getAPI(c, "My Speed Test");
+        Date endTime = null;
+        //MeasurementTask task = null;
+
+        ArrayList<MeasurementTask> taskList = new ArrayList<MeasurementTask>();
+        List <Pair<String, String>> slist=ServerUtil.getDnsTargets();
+            /*if(slist != null && slist.size() != 0 ) {
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException e) {}
+            }*/
+        count = slist.size();
+        for (int i=0; i < count; i++)
+        {
+            String target=((Pair<String, String>)slist.get(i)).first;
+            String server=((Pair<String, String>)slist.get(i)).second;
+
+            DnsHandler dnsHandler=new DnsHandler(context);
+            DnsLookupTask task = new DnsLookupTask(target, server, dnsHandler);
+            dnsThreadPool.execute(task);
+        }
+        //params.put("target", "www.youtube.com");
+        //params.put("server", "8.8.8.8");
+
+        //task = mobilyzer.createTask(API.TaskType.DNSLOOKUP, Calendar.getInstance().getTime(),
+        //        endTime, 60, 1, MeasurementTask.USER_PRIORITY, 1, params);
+        //mobilyzer.submitTask(task);
+
+        //} catch (MeasurementError e) {
+        //    e.printStackTrace();
+        //}
+    }
+
+    private class DnsHandler extends Handler {
+
+        Context context;
+
+        public DnsHandler(Context context)
+        {
+            this.context = context;
+            measurement = new Measurement(context, false);
+        }
+
+        @Override
+        public void handleMessage(Message msg) {
+            String target = msg.getData().getString("target");
+            String server = msg.getData().getString("server");
+            String address = msg.getData().getString("address");
+            String totaltime = msg.getData().getString("totaltime");
+            String success = msg.getData().getString("success");
+            String totalattempts = msg.getData().getString("totalattempts");
+            String error = msg.getData().getString("error");
+            Log.i("DnsLookupTask", "Target:"+target+" Server:"+server+" Address:"+address+" TotalTime:"+totaltime+" Success:"+success+
+                    " TotalAttempts: "+totalattempts+" Error:"+error);
+
+            dnses.add(new Dns(target, server,address, error, success, totaltime, totalattempts));
+            onComplete();
+        }
+    }
+
+    private void onComplete() {
+        //Log.d("DnsManager", "count:"+count+" dnses.size:"+dnses.size()+" ********************");
+        if(dnses.size() >= count) {
+            measurement.setDnses(dnses);
+            sendMeasurement(measurement);
+        }
+    }
+
+    public void sendMeasurement(final Measurement measurement) {
+        if(Constants.DEBUG) {
```

Comment:
If I understand this section correctly, you are chopping up the log data to ensure that we stay within the bounds on the max string size for a log entry?